### DISCUSSION TITLE
G Theory Setup Question

### DISCUSSION THREAD

**Discussion by pstur1** [G-String-Legacy/G_String#18] [Q&A]
Hi,

I am working on a G-theory analysis, and was hoping that someone would be able to answer my questions.  The data that I have consists of faculty evaluations of student performance in clerkships.  The issue is that different students were evaluated a different number of times in the various clerkships.  For example, student one was evaluated 33 times total, 9 times in clerkship 1, two times in clerkship two, six times in clerkship three, was not evaluated at all in clerkship four, seven times in clerkship five, and eight times in clerkship six.  Student two was evaluated 27 times and the number of times they were evaluated in the different clerkships is also similarly random.  The data for the other students (approximately 100) is similar.  I also have data on who rated each student, and the location that each clerkship took place.

So far, I have conducted an analysis in SAS on the entire data set in which I was able to calculate the variance components, which successfully answered my primary research question (most of the variance was due to the rater rather than other factors).  I also deleted data to the point where I had 12 evaluations for each student, two in each clerkship.  I analyzed the data both in SAS and G String, and was able to successfully conduct the generalizability and dependability studies on the smaller data set.

What I would like to be able to do in G String is calculate the variance components and the generalizability coefficient for the entire data set.  Is this possible, and if so, how would I set up the data file in order to conduct the analysis?

FYI, my SAS data file for the first few students is attached.

Thanks for any suggestions!

[first three cases.xlsx](https://github.com/G-String-Legacy/G_String/files/8709571/first.three.cases.xlsx)

**Comment by stevenarild** [comment]
It is a highly complex dataset you have and dealing with unbalanced designs in G analysis can be tricky and there are many potential pitfalls. As a clinician/medical educator, I have found the need to ally myself with a statistician for those cases. I don't think there is an easy way to do an analysis of the entire dataset at once. Rather, I would recommend you think about your research question and focus the analysis on what you want to know to reduce complexity and enable analysis.

I asked Ralph Bloch (who developed the G_String software) and he also recommends considering the primary question and doing analysis accordingly: for example simply testing for rep:student might be sufficient in high stakes student rating, or if focus is on rotation then to analyze rotation rep:student. The number of assessors seems to be too high to include as a facet. The nesting might require you to restructure the dataset, so again, considering what you want to know is important to avoid redoing the datasets multiple times.

Some of the others on here might be able to pitch in with their experience as well.

Best,
Steven
